scripts/src/javatests/org/oppia/android/scripts/kdoc/KDocCheckTest.kt

```diff
@@ -0,0 +1,487 @@
+package org.oppia.android.scripts.kdoc
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+
+/** Tests for [KDocCheck]. */
+class KDocCheckTest {
```

Comment:
Other tests that I think are worth adding:
- A block comment instead of a KDoc (to ensure KDocs are specifically checked).
- Inject primary constructor (just to make sure nothing special is triggered in that situation)
- Object vals
- Constants
- Companion objects
- Nested interfaces
- Nested annotations
- All other special case annotations not already tested below (all of them ought to be checked)
- The special case of the exemptions list having a redundant file in it (see my comment above)

Reply:
Added.